fix: Poll roles, remove canGetPermissions permission

mapmeld opened the pull request:
## Description

I'm trying to resolve ongoing issues with the Workspace Members page. Currently on an RBAC/EE server, the row has the current role's number ID and no selectable roles.

I am making two changes here:
- Using `useFetchKnownRoles` to populate the roles dropdown in Members page.
- The above change also helps fix the Add Member modal. Any role can be assigned to a workspace member.
- Change/remove workspace member role is now functioning from API change
- ~Removing `canGetPermissions` since it was an alias for the Assign Roles permission, and we should be able to view permissions / members of the workspace even if we can't assign roles.~

## Test Plan

- Use a determined-ee environment to fully test API and client
- Log in as admin
- Create a workspace. The table in the members tab should include admin ~(as admin, you are not shown)~
- In the members tab, click the Add Members button, and add 'determined' to the workspace with a Viewer role. 
- Confirm that determined is shown in the table now and after reloading.
- Use the determined row's dropdown to change role to Editor. Confirm that the new role is shown now and after reloading the page.
- Log in as determined, and confirm that you can view the workspace (you will not see members listed).
- Log in as admin again. In the members table, on the right end of the 'determined' row, you should have a three-dots menu. Click that button to open a dropdown and click 'Remove'. In this modal, confirm removing this role from 'determined'.

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

julian-determined-ai commented:
**question:** Is the following point being actively discussed?
> This doesn't filter the roles, so maybe this needs to reduce it to workspace-specific roles, I am not sure.

The `canGetPermissions` change makes sense to me. I am fine approving this PR if there will be a follow up to correctly populate roles. I am just unsure if more work needs to be done to make sure we are correctly populating roles.

mapmeld commented:
New fixes for workspace members:
- Add Member modal was sending `g_{id}` or `u_{id}` to API; updated to send actual number ID
- Each row dropdown can change or remove the member's role. Both actions include removing a role, and `scopeWorkspaceId` was sent in the wrong level of the body. Updated this call.
- Made `getAssignedRole` return a `V1RoleAssignment` object so UI knows the scope.
-- I don't believe this is an actual issue because the Admin's global role is not appearing here at all

Ongoing issues:
- Remove Role modal does a sort of hide/show/hide flash.
- All roles are still available in the dropdown, even though ClusterAdmin does not sound right there
- We talked on Slack about in the future, a user with a global assignment should appear on the page in its own table/section

julian-determined-ai commented:
@mapmeld I wanted to double check ` The table in the members tab should be empty (as admin, you are not shown)` 

When I create a new workspace I do see myself in the table:

![Screen Shot 2022-10-24 at 3 14 11 PM](https://user-images.githubusercontent.com/103522725/197619997-50c0eee8-3164-4abf-99d0-af749b18e951.png)

It makes sense to me that the `WorkspaceAdmin` would be in the table so wanted to clarify.

mapmeld commented:
@julian-determined-ai you're right, I had removed the permission and not tried it in a fresh new workspace. I'll update the test info

julian-determined-ai commented:
Looks like there is an issue with removing a member. The UI gets stuck in an infinite loop for a period of time. 

https://user-images.githubusercontent.com/103522725/197623485-9e3c2324-1d36-449c-95fe-f59d6c384e8f.mov

julian-determined-ai commented:
@mapmeld I had picked up https://determinedai.atlassian.net/browse/WEB-552 which cover fixing the add/edit members functionality. Your PR fixes the issue with adding members. 

If we wanted to keep the scope of this PR small and plan for me to fix the issues with editing functionality and get this shipped that works for me. Let me know your preference.
